src/core/ui.js

```diff
@@ -60,25 +61,59 @@ const respecPill = hyperHTML`<button id='respec-pill' disabled>ReSpec</button>`;
 respecUI.appendChild(respecPill);
 respecPill.addEventListener("click", e => {
   e.stopPropagation();
-  if (menu.hidden) {
-    menu.classList.remove("respec-hidden");
-    menu.classList.add("respec-visible");
-  } else {
-    menu.classList.add("respec-hidden");
-    menu.classList.remove("respec-visible");
-  }
   respecPill.setAttribute("aria-expanded", String(menu.hidden));
-  menu.hidden = !menu.hidden;
+  toggleMenu();
+  menu.querySelector("li:first-child *").focus();
 });
+
 document.documentElement.addEventListener("click", () => {
   if (!menu.hidden) {
-    menu.classList.remove("respec-visible");
-    menu.classList.add("respec-hidden");
-    menu.hidden = true;
+    toggleMenu();
   }
 });
 respecUI.appendChild(menu);
 
+menu.addEventListener("keydown", e => {
+  if (e.key === "Escape" && !menu.hidden) {
+    respecPill.setAttribute("aria-expanded", String(menu.hidden));
+    toggleMenu();
+    respecPill.focus();
+  }
+});
+
+function toggleMenu() {
+  menu.classList.toggle("respec-hidden");
+  menu.classList.toggle("respec-visible");
+  menu.hidden = !menu.hidden;
+}
+
+// Code adapted from https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element
+function trapFocus(element) {
+  const focusableEls = element.querySelectorAll(
+    "a[href]:not([disabled]), button:not([disabled]), textarea:not([disabled]), input:not([disabled]), select:not([disabled])"
+  );
+  const firstFocusableEl = focusableEls[0];
+  const lastFocusableEl = focusableEls[focusableEls.length - 1];
+
+  element.addEventListener("keydown", e => {
+    if (e.key !== "Tab") {
+      return;
+    }
+    // shift + tab
```

Comment:
Not liking the repletion here (no big deal though)
Can probably do something like?
``` js
let focusableEl = null;
if (e.shiftKey && document.activeElement === firstFocusableEl) {
  focusableEl = lastFocusableEl;
} else if (!e.shiftKey && document.activeElement === lastFocusableEl) {
  focusableEl = firstFocusableEl;
}
if (focusableEl) {
  focusableEl.focus();
  e.preventDefault();
}
```

Or a slightly cryptic version:
``` js
let focusableEl = null;
const focusEls = [firstFocusableEl, lastFocusableEl];
if (document.activeElement === focusEls[Number(e.shiftKey)]) {
  focusableEl = focusEls[Number(!e.shiftKey)];
}
if (focusableEl) {
  focusableEl.focus();
  e.preventDefault();
}
```

Reply:
Let's not be cryptic 😂

Reply:
Is it fine presently or should I revert the changes

Reply:
Keep them.